index.js

```diff
@@ -14,7 +14,12 @@ const slowDownPlugin = async (fastify, settings) => {
   fastify.addHook('onClose', () => store.close())
 
   fastify.addHook('onRequest', async (req, reply) => {
-    const requestCounter = store.incrementOnKey(options.keyGenerator(req))
+    if (options.skip(req, reply)) {
```

Comment:
this is a hot path, meaning that it's executed for all requests. let's avoid an unnecessary function invocation by removing the default value of the option and invoking this conditionally

```suggestion
    if (options.skip?.(req, reply)) {
```